### DISCUSSION TITLE
Potential Bug in YALMIP: No Error for Dimension Mismatch in Equality Constraints

### DISCUSSION THREAD

**Discussion by keeping118** [yalmip/YALMIP#1441] [Possible bug or missing functionality] [answered]
Dear professor.
        I have encountered a potential bug while using YALMIP, and I would appreciate your insights. When the dimensions of the left-hand side and right-hand side of an equation do not match, YALMIP still proceeds with the assignment and does not throw an error. Here is a simplified example:

```
clc;clear 
T=2;
b=[-1 -2]';
a=sdpvar(1,1);
Constraints = [];
Constraints = [Constraints, a==b];
ops = sdpsettings('solver','Gurobi+','verbose',2,'gurobi.ResultFile','MyFile.lp');
result=optimize(Constraints,[],ops);  
[model,recoverymodel] = export(Constraints, [], sdpsettings('solver','GUROBI'));
IIS = gurobi_iis(model);
gurobi_write(model,'abc1.lp');
value(a)
```
In this example, a is a 1x1 variable, and b is a 2x1 vector. This should ideally trigger a dimension mismatch error, but YALMIP does not report any error, although the model is marked infeasible. Upon inspecting the output file MyFile.lp, I noticed that YALMIP translates this into two equality constraints:
```
\ Model matlab
\ LP format - for model browsing. Use MPS format to capture full model detail.
Minimize

Subject To
 R0: C0 = -1
 R1: C0 = -2
Bounds
 C0 free
End
```
Should YALMIP throw an error when such dimension mismatches occur, or am I misunderstanding how this is handled? Your guidance would be greatly appreciated.

**Comment by johanlofberg** [comment, marked as answer]
That's simply MATLAB syntax, so called singleton expansion

```
>> [1;2;3]-5

ans =

    -4
    -3
    -2

```

**Reply by keeping118** [reply]
It was my misunderstanding; dimension mismatches occur only between matrices. Thank you for your answer.
